openmindat/mindat_api.py

```diff
@@ -214,22 +214,36 @@ def get_mindat_json(self, PARAM_DICT, END_POINT, VERBOSE = 2):
         end_point = END_POINT
 
         # Retrieve the first page of data
-        response = requests.get(self.MINDAT_API_URL+ "/" + end_point + "/",
-                        params=params,
-                        headers=self._headers)
-        try:
-            response_json = response.json()
-            result_data = response_json["results"]
-        except KeyError:
-            # This error indicates the result only has one page
-            # We will convert the result data into a list for consistency
-            result_data = [response_json]
-        except TypeError:
-            # This error indicates the result data is a list instead of a dict
-            # We will pass the response result directly
-            result_data = response_json
-        except:
-            raise ValueError(str(response.reason))
+        while True:
+            response = requests.get(self.MINDAT_API_URL+ "/" + end_point + "/",
+                            params=params,
+                            headers=self._headers)
+            
+            if len(response.url) > 4097:
+                raise ValueError("Search query to big, reduce the size of the search and try again.")
+            
+            try:
+                response_json = response.json()
+                result_data = response_json["results"]
+                break
+            except KeyError:
+                # This error indicates the result only has one page
+                # We will convert the result data into a list for consistency
+                result_data = [response_json]
+                break
+            except TypeError:
+                # This error indicates the result data is a list instead of a dict
+                # We will pass the response result directly
+                result_data = response_json
+                break
+            except ValueError:
```

Comment:
The logic here seems weird. Did you mean the gateway timeout error will throw a `ValueError`? If so, I still recommend handling this exception differently: instead of wrapping the entire retrieval in a `while Ture` loop, use a for loop in range(3) to avoid adding `break`s everywhere. This might also remove the requirement of checking if `page_size`'s value is less than 200: raising an error when `page_size` is less than 200 appears weird. If I need clarification on the code, please let me know. Thanks.